src/Avalonia.Controls/VirtualizingStackPanel.cs

```diff
@@ -185,13 +238,15 @@ protected override Size MeasureOverride(Size availableSize)
 
                 // If there is a focused element is outside the visible viewport (i.e.
                 // _focusedElement is non-null), ensure it's measured.
-                _focusedElement?.Measure(availableSize);
+                MeasureElement(_focusedElement, availableSize);
 
                 return CalculateDesiredSize(orientation, items.Count, viewport);
             }
             finally
             {
+                _cacheElementMeasurements = false;
```

Comment:
Took me a little which to understand why `_cacheElementMeasurements` is reset here, but if I remove this line, then `Focused_Container_Is_Positioned_Correctly_when_Container_Size_Change_Causes_It_To_Be_Moved_Into_Visible_Viewport` fails, so I *think* that the reason for it is that it's only set to true after a viewport change due to scrolling, and if a measure is invoked for any other reason then it needs to be false? If so might be worth adding a quick comment here to explain it (as the rest of the code is so well commented).

Reply:
yeah that is exactly the idea.
My problem was, that I wanted to make scrolling fast in a virtualizingstackpanel (as fast as it is in a normal stackpanel) so I tried caching as much as possible.

Doesn't a call to "InvalidateMeasure" invalidate all measurements down the visual tree?  At least I thought it did.
One reason why the normal "VirtualizingStackPanel" is so slow (for complex item layouts) is that it measures everything anytime a new item is shown.
So I _assume_ the caching you are talking about is at least not working? (if it _is_, I am happy to remove that code and the complexity that comes with it)